### PR TITLE
[DOC] Add intro material for Pyroscope

### PR DESCRIPTION
**Pull request by knylander-grafana**
Draft: Adds and organizes the intro material for Pyroscope.  The best way to view this PR is to check it out and build locally. 

To do: 
* Reduce duplicated content
* Consoliidate the content for the UI on one page
* Review page suggestions in the individual issues and verify that they are addressed. 

Fixes: #2736 #2770

Related to #2776 #2770 #2794

### CODE REVIEW

File: docs/sources/_index.md [hunk 1]
```diff
@@ -16,16 +16,33 @@ keywords:
...
+
+Continuous profiling is more than just a performance analysis tool.
+It's a crucial component in modern software development and operations.
+It goes past traditional profiling techniques by providing ongoing, in-depth insights into application performance.
+
+Continuous profiling goes past the ephemeral, localized nature of traditional profiling (which historically has been more similar to `console.log` or `print statement` debugging) to a structured, centralized approach allows for effective use in production environments.
+Put more simply, Pyroscope takes you from a bunch of flamegraph files on your desktop to a database where you can query and analyze production data in a structured way.
+
+{{% admonition type="tip" %}}
+To learn more about continuous profiling, read [the Introduction section]({{< relref "introduction" >}}).
```
Comment (knylander-grafana) [inline docs/sources/_index.md:34]:
```suggestion
To learn more about continuous profiling, read [the Introduction section]({{< relref "./introduction" >}}).
```

File: docs/sources/upgrade-guide/_index.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
-title: "Upgrade to Grafana Pyroscope v1.0"
-menuTitle: "Upgrade to v1.0"
-description: "Upgrading to Grafana Pyroscope v1.0"
+title: Upgrade your Pyroscope installation
```
Comment (knylander-grafana) [inline docs/sources/upgrade-guide/_index.md:2]:
```suggestion
title: Upgrade your Grafana Pyroscope installation
```

**Review by Rperry2174 — APPROVED** [review]
Looks great! So much better :)

**Comment by knylander-grafana** [comment]
@mkrummz is willing to help with content, so I'm going to merge this PR. She'll be working on the Ingest and analyze profile data section.

**Pull request merged**